docs/src/modules/java/pages/eventsourced.adoc

```diff
@@ -3,108 +3,137 @@
+<6> Messages describe the return value for our API. For methods that don't have return values, we use `google.protobuf.Empty`.
+<7> The service descriptor shows the API of the entity. It lists the methods a client can use to issue Commands to the entity.
+<8> The protobuf option `(akkaserverless.service)`  is specific to code-generation as provided by the Akka Serverless Maven plugin and points to the protobuf definition `ShoppingCart` we've seen above (in the `com.example.shoppingcart.domain` package).
 
-== Constructing
+== Implementing behavior
 
-Akka Serverless constructs instances of the Event Sourced Entity class on demand. The constructor below shows how the unique ID of the entity is injected and stored in link:{attachmentsdir}/api/com/akkaserverless/javasdk/EntityId.html[`entityId` {tab-icon}, window="new"].
+A Event Sourced Entity implementation is a Java class where you define how each command is implemented. The class `ShoppingCart` gets generated for us based on the `shoppingcart_api.proto` and `shoppingcart_domain.proto` definitions. Once the `ShoppingCart.java` file exist, it is not overwritten, so you can freely add logic to it. `ShoppingCart` extends the generated class `AbstractShoppingCart` which we're not supposed to change as it gets regenerated in case we update the protobuf descriptors. `AbstractShoppingCart` contains all method signatures corresponding to the API of the service. If you change the API you will see compilation errors in the `ShoppingCart` class and you have to implement the methods required by `AbstractShoppingCart`.
 
-[source,java,indent=4]
+[source,java]
+.src/main/java/com/example/shoppingcart/domain/ShoppingCart.java
 ----
-include::example$java-eventing-shopping-cart/src/main/java/shopping/cart/domain/ShoppingCartEntity.java[tag=constructor]
+include::example$java-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/domain/ShoppingCart.java[tag=class]
 ----
+<1> Extends the generated `AbstractShoppingCart`, which extends link:{attachmentsdir}/api/com/akkaserverless/javasdk/eventsourcedentity/EventSourcedEntity.html[`EventSourcedEntity` {tab-icon}, window="new"].
+<2> Defines the initial, empty, state that is used before any updates.
+
+We need to implement all methods our Event Sourced Entity offers as https://developer.lightbend.com/docs/akka-serverless/reference/glossary.html#command_handler[_command handlers_].
+
+The code-generation will generate an implementation class with an initial empty implementation which we'll discuss below.
 
-== Handling commands
+Command handlers show in the implementation class as methods that override abstract methods from `AbstractShoppingCart`. The methods take the current state as the first parameter and the request message as the second parameter. They return an `Effect`, which describes next processing actions, such as emitting events and sending a reply.
```

Comment:
Not sure `show` in `Command handlers show in the implementation class as methods ...` reads that well. Maybe `are implemented` is more direct, even with `implementation class`. Or could be `are required` or similar.

```suggestion
Command handlers are implemented in the implementation class as methods that override abstract methods from `AbstractShoppingCart`. The methods take the current state as the first parameter and the request message as the second parameter. They return an `Effect`, which describes next processing actions, such as emitting events and sending a reply.
```

Reply:
I agree, must be similar in value entity docs. "implemented in the implementation" is somewhat funny. I'll see if I can rephrase that.